main.py

```diff
@@ -0,0 +1,449 @@
+#!/usr/bin/env python
+
+import time
+import os
+import random
+import threading
+import argparse
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy as sc
+import cv2
+from collections import namedtuple
+import torch  
+from torch.autograd import Variable
+from robot import Robot
+from trainer import Trainer
+from logger import Logger
+import utils
+
+
+def main(args):
+
+
+    # --------------- Setup options ---------------
+    is_sim = args.is_sim # Run in simulation?
+    obj_mesh_dir = os.path.abspath(args.obj_mesh_dir) if is_sim else None # Directory containing 3D mesh files (.obj) of objects to be added to simulation
+    num_obj = args.num_obj if is_sim else None # Number of objects to add to simulation
+    tcp_host_ip = args.tcp_host_ip if not is_sim else None # IP and port to robot arm as TCP client (UR5)
+    tcp_port = args.tcp_port if not is_sim else None
+    rtc_host_ip = args.rtc_host_ip if not is_sim else None # IP and port to robot arm as real-time client (UR5)
+    rtc_port = args.rtc_port if not is_sim else None
+    if is_sim:
+        workspace_limits = np.asarray([[-0.724, -0.276], [-0.224, 0.224], [-0.0001, 0.4]]) # Cols: min max, Rows: x y z (define workspace limits in robot coordinates)
+    else:
+        workspace_limits = np.asarray([[0.3, 0.748], [-0.224, 0.224], [-0.255, -0.1]]) # Cols: min max, Rows: x y z (define workspace limits in robot coordinates)
+    heightmap_resolution = args.heightmap_resolution # Meters per pixel of heightmap
+    random_seed = args.random_seed
+    force_cpu = args.force_cpu
+
+    # ------------- Algorithm options -------------
+    method = args.method # 'reactive' (supervised learning) or 'reinforcement' (reinforcement learning ie Q-learning)
```

Comment:
Here is where they mention the difference between reactive and reinforcement learning algorithms